File: kotlin/euler/src/main/kotlin/com/rbleggi/euler/P29.kt
package com.rbleggi.euler

import java.math.BigInteger

fun mainP29() {
    val terms = (2..100).flatMap { a ->
        (2..100).map { b -> BigInteger.valueOf(a.toLong()).pow(b) }
    }.toSet()

    println("Number of distinct terms for 2 <= a <= 100 and 2 <= b <= 100: ${terms.size}")
}

File: scala-3/euler/src/main/scala/com/rbleggi/euler/P29.scala
package com.rbleggi.euler

@main def mainP29(): Unit = {
  val terms = (for {
    a <- 2 to 100
    b <- 2 to 100
  } yield BigInt(a).pow(b)).toSet

  println(s"Number of distinct terms for 2 <= a <= 100 and 2 <= b <= 100: ${terms.size}")
}
